### PR TITLE
Improve handling for shared values in Blazor Server

### PR DESCRIPTION
**Pull request by captainsafia**
**Changes in this PR**
- Throws `InvalidOperationException` in RendererRegistry APIs are being invoked from non-WASM runtimes
- Makes more values `readonly`

The original issue pointed out that the `EventArgs.Empty` value is shared and we need to ensure that it is immutable. From what I can see, it is already set as a read-only property so we shouldn't expect any unwarranted mutations here.

Addresses #11849

### CODE REVIEW

File: src/Components/Blazor/Blazor/src/Rendering/RendererRegistry.cs
Comment (SteveSandersonMS) [inline src/Components/Blazor/Blazor/src/Rendering/RendererRegistry.cs:19]: In the `blazor-wasm` branch the platform name is now `BROWSER` instead of `WEBASSEMBLY` so this will break when we merge.

However I agree your change is right as far as the `master` branch is concerned :) We'll just have to update it when we merge things.

I'm only raising this point to spread awareness about this change to the platform name.

File: src/Components/Blazor/Blazor/src/Rendering/RendererRegistry.cs
Comment (SteveSandersonMS) [inline src/Components/Blazor/Blazor/src/Rendering/RendererRegistry.cs:51]: Just a thought, which I don't mind whether you do anything with or not :) Instead of having multiple checks for `isWebAssembly` that run each time there are calls, an alternative would be changing the static constructor so that the static `_renderers` dictionary doesn't even get instantiated in non-wasm cases.

Then there would be no need to check `isWebAssembly` in each of these methods - they would just nullref if they were called on the wrong platform (and they never should be called like this - it could only happen if we introduced a bug since it's `internal` anyway - so a nullref is a reasonable level of error handling).

**Comment by SteveSandersonMS** [comment]
> The original issue pointed out that the EventArgs.Empty value is shared and we need to ensure that it is immutable. From what I can see, it is already set as a read-only property so we shouldn't expect any unwarranted mutations here.

You're absolutely right - that's not going to be a problem :)

**Review by SteveSandersonMS — APPROVED** [review]
Nice!

**Comment by pranavkm** [comment]
My suggestion was to wait until we merge blazor-wasm -> 3.1 -> master. It's going to be added complexity if we have to resolve merge conflicts since a lot of these types have changed in the wasm branch.

File: src/Components/Blazor/Blazor/src/Rendering/RendererRegistry.cs
Comment (captainsafia) [reply src/Components/Blazor/Blazor/src/Rendering/RendererRegistry.cs:51]: Yep, I can make this change.

**Comment by captainsafia** [comment]
> My suggestion was to wait until we merge blazor-wasm -> 3.1 -> master. It's going to be added complexity if we have to resolve merge conflicts since a lot of these types have changed in the wasm branch.

Yep, I'm in agreement with this. I opened a PR just so we have something to merge in once master has been updated with the blazor-wasm changes. I'll keep this in draft mode until then but the reviews are good to square away all the feedback.

**Comment by captainsafia** [comment]
Merging this now.

**Pull request merged**
